### ISSUE TITLE
[Critical] Role: The element's role is missing or incorrect

### ISSUE THREAD

**Issue by ntallman** [APTrust/registry#110]
## Issue Details

| | |
|---|---|
| **Source Issue ID** | 87918 |
| **Checkpoint** | 4.1 Compatible › Custom Controls (4.1.2.b) |
| **Impact** | Critical |
| **Standards** | wcag2a, wcag2aa, wcag21a, wcag21aa, wcag22a, wcag22aa |
| **Test Area** | 06 Alerts |
| **Test Case Name** | AP Trust Registry (Demo) |
| **URL** | <https://demo.aptrust.org/alerts> |
| **Environment** | Production |
| **Platform** | Windows | Chrome | Firefox |
| **Assistive Technology** | JAWS | NVDA | Magnifier |
| **Method** | Manual |
| **Digital Asset Type** | Desktop Web |
| **Release** | 4.2026 |
| **Product** | Website |
| **Date Reported** | 05/17/26 at 1:28 pm |
| **Unit Type** | Page |
| **Issue Type** | Accessibility |
| **Reported By** | kim.casey@ablr360.com |

## Description

The element's role is missing or is not appropriate for the element's function.
The Mark as Read/Unread function in the Read Status column is marked up as a link, which would imply that it takes you to another page or resource, but it acts as a button to launch a dialog.

## Recommendation

Rule : 
The name, role, value, states, and properties of user interface components MUST be programmatically determinable by assistive technologies.

How To Fix: 
Fix this issue by only using appropriate, valid ARIA roles per the ARIA Recommendation document and the ARIA Authoring Practices document (see References below).

Reference: 
Buttons vs. Links: https://accessibilitymadeeasy.org/buttons-vs-links-making-them-accessible-and-effective/
Button vs Link: Choosing the right one: https://vispero.com/resources/link-vs-button-choosing-the-right-element-for-the-right-job/

Background: 
Every user interface control must have a role to convey what type of control it is for screen reader and other assistive technology users. Native HTML elements - such as <button>, <a>, <input>, <select> - already have a role, so nothing more needs to be done. If you create a custom version of a native HTML element or a custom control or widget that does not have a native HTML equivalent, you must add the relevant role(s) using ARIA as well as expected keyboard interactions.

## Source Code

```
<a href="#">
            <span data-name="alertReadIcon" class="material-icons is-gold" aria-hidden="true">mail</span>
            <span class="is-sr-only">Mark as Read</span>
          </a>
```

## Screenshots

![06_label meaningful](https://raw.githubusercontent.com/APTrust/registry/master/docs/accessibility-audit/screenshots/87918/06_label_meaningful.png)
